xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -953,6 +953,14 @@ void CVideoPlayer::OpenDefaultStreams(bool reset)
   if (!std::dynamic_pointer_cast<CDVDInputStreamNavigator>(m_pInputStream) || m_playerOptions.state.empty())
     SetSubtitleVisibleInternal(visible); // only set subtitle visibility if state not stored by dvd navigator, because navigator will restore it (if visible)
 
+  if (!visible && m_pInputStream && m_pInputStream->IsStreamType(DVDSTREAM_TYPE_ADDON))
```

Comment:
This probably makes more sense if added in SetSubtitleVisibleInternal

Reply:
initially i was add the code on SetSubtitleVisibleInternal but i noticed that SetSubtitleVisibleInternal  could be called by different methods so i was not sure, but if you say it is ok then i move on it

Reply:
i moved on SetSubtitleVisibleInternal method,
but i havent removed `!bVisible &&` condition

before i want pointing out/discuss one thing

atm `OpenStream` callback implicitly means `EnableStream` with "true" param value (despite this callback atm dont exist),
 `EnableStream` with false value callback, is done only with CloseStream.

now if i remove `!bVisible &&` condition will make inconsistent callbacks,
just for the subtitle stream the `EnableStream` is called and not for all others stream types

Therefore, imo it would be preferable to have `EnableStream` callbacks for all stream types,
so that with `OpenStream` callback `EnableStream` callback is not more implicit, but explicit,
imo we should add to the end of operations of `CVideoPlayer::OpenStream` method an `EnableStream` callback for InputStream by excluding subtitle stream because managed by SetSubtitleVisibleInternal

Reply:
i have reworked things to have consistents `EnableStream` callbacks between all stream types,
please rewiew again, i will squash commits later

Reply:
ok i reworked again, decoupled things (removed the previous temp commit changes)